elements/lisk-p2p/test/integration/p2p.ts

```diff
@@ -1847,4 +1847,130 @@ describe('Integration tests for P2P library', () => {
+					maxInboundConnections: DEFAULT_MAX_INBOUND_CONNECTIONS,
+					minimumPeerDiscoveryThreshold: 1,
+					nodeInfo: {
+						wsPort: nodePort,
+						nethash:
+							'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
+						version: '1.0.1',
+						protocolVersion: '1.1',
+						minVersion: '1.0.0',
+						os: platform(),
+						height: 0,
+						broadhash:
+							'2768b267ae621a9ed3b3034e2e8a1bed40895c621bbb1bbd613d92b9d24e54b5',
+						nonce: `O2wTkjqplHII${nodePort}`,
+					},
+				});
+			});
+			// Launch nodes one at a time with a delay between each launch.
+			for (const p2p of p2pNodeList) {
+				await p2p.start();
+			}
+			await wait(200);
+		});
+
+		afterEach(async () => {
+			await Promise.all(
+				p2pNodeList
+					.filter(p2p => p2p.isActive)
+					.map(async p2p => await p2p.stop()),
+			);
+			await wait(100);
+		});
+
+		it('should return list of peers with at most the minimum discovery threshold', async () => {
+			const firstP2PNode = p2pNodeList[0];
+			const { newPeers } = firstP2PNode.getNetworkStatus();
+			expect(newPeers.length).to.eql(1);
+		});
+	});
+
+	describe('Connected network: Maximum peer discovery response size is set to 3', () => {
+		beforeEach(async () => {
+			p2pNodeList = [...new Array(NETWORK_PEER_COUNT).keys()].map(index => {
+				// Each node will have the previous node in the sequence as a seed peer except the first node.
+				const seedPeers =
+					index === 0
+						? []
+						: [
+								{
+									ipAddress: '127.0.0.1',
+									wsPort: NETWORK_START_PORT + index - 1,
+								},
+						  ];
+
+				const nodePort = NETWORK_START_PORT + index;
+
+				return new P2P({
+					connectTimeout: 10000,
+					ackTimeout: 200,
+					seedPeers,
+					wsEngine: 'ws',
+					populatorInterval: 10000,
+					maxOutboundConnections: DEFAULT_MAX_OUTBOUND_CONNECTIONS,
+					maxInboundConnections: DEFAULT_MAX_INBOUND_CONNECTIONS,
+					maximumPeerDiscoveryResponseSize: 3,
+					nodeInfo: {
+						wsPort: nodePort,
+						nethash:
+							'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
+						version: '1.0.1',
+						protocolVersion: '1.1',
+						minVersion: '1.0.0',
+						os: platform(),
+						height: 0,
+						broadhash:
+							'2768b267ae621a9ed3b3034e2e8a1bed40895c621bbb1bbd613d92b9d24e54b5',
+						nonce: `O2wTkjqplHII${nodePort}`,
+					},
+				});
+			});
+			// Launch nodes one at a time with a delay between each launch.
+			for (const p2p of p2pNodeList) {
+				await p2p.start();
+			}
+			await wait(200);
+		});
+
+		afterEach(async () => {
+			await Promise.all(
+				p2pNodeList
+					.filter(p2p => p2p.isActive)
+					.map(async p2p => await p2p.stop()),
+			);
+			await wait(100);
+		});
+
+		it('should return list of peers with less than maximum discovery response size', async () => {
+			const firstP2PNode = p2pNodeList[0];
+			const { newPeers } = firstP2PNode.getNetworkStatus();
+			expect(newPeers.length).to.be.lessThan(3);
```

Comment:
As above, convert this to a variable. You can reuse it in the title and the P2PConfig object.